specifyweb/frontend/js_src/lib/components/Atoms/Form.tsx

```diff
@@ -161,46 +166,33 @@ export const Input = {
     'Input.Generic',
     'input',
     `${className.notTouchedInput} w-full`,
-    ({
-      onValueChange,
-      onDatePaste: handleDatePaste,
-      isReadOnly,
-      ...props
-    }) => ({
+    ({ onValueChange, isReadOnly, ...props }) => ({
       ...props,
       ...withHandleBlur(props.onBlur),
       onChange(event): void {
         onValueChange?.((event.target as HTMLInputElement).value);
         props.onChange?.(event);
       },
-      onPaste(event): void {
-        const target = event.target as HTMLInputElement;
-        // Ignore date paste if there is some selected text
-        const hasSelectedRegion = target.selectionEnd !== target.selectionStart;
-        // Handle pasting dates into input[type="date"] and [type="month"]
-        if (typeof handleDatePaste === 'function' && !hasSelectedRegion) {
-          const input =
-            target.tagName === 'INPUT'
-              ? target
-              : target.getElementsByTagName('input')[0];
-          const initialType = input.type;
+      onDoubleClick(event): void {
+        const input = event.target as HTMLInputElement;
+        if (input.type === 'date') {
           input.type = 'text';
-          try {
-            const value =
-              // @ts-expect-error globalThis.clipboardData does not have typings
-              (event.clipboardData ?? globalThis.clipboardData).getData(
-                'text/plain'
-              );
-            handleDatePaste(value);
-          } catch (error: unknown) {
-            console.error(error);
+          const parsed = parseDate('full', input.value);
+          if (parsed.isValid()) input.value = parsed.format(fullDateFormat());
```

Comment:
Can someone please test if this works in the query builder
(i.e, whether it queries on the correct date). We might have to change `fullDateFormat()` back to `databaseFieldFormat` after all)